packages/@glimmer/runtime/lib/vm/attributes/dynamic.ts

```diff
@@ -81,18 +81,20 @@ export class SimpleDynamicAttribute extends DynamicAttribute {
 }
 
 export class DefaultDynamicProperty extends DynamicAttribute {
+  value: Opaque;
   set(dom: ElementBuilder, value: Opaque, _env: Environment): void {
     if (value !== null && value !== undefined) {
       let { name } = this.attribute;
+      this.value = value;
       dom.__setProperty(name, value);
     }
   }
 
   update(value: Opaque, _env: Environment): void {
     let { element, name } = this.attribute;
 
-    if (element[name] !== value) {
-      element[name] = value;
+    if (this.value !== value) {
+      element[name] = this.value = value;
     }
 
     if (value === null || value === undefined) {
```

Comment:
Seems like we probably want to avoid calling `this.removeAttribute()` if the value is unchanged as well, right?